src/main/kotlin/no/nav/familie/ef/sak/vilkår/gjenbruk/GjenbrukVilkårService.kt

```diff
@@ -0,0 +1,162 @@
+package no.nav.familie.ef.sak.vilkår.gjenbruk
+
+import no.nav.familie.ef.sak.barn.BarnService
+import no.nav.familie.ef.sak.barn.BehandlingBarn
+import no.nav.familie.ef.sak.behandling.BehandlingService
+import no.nav.familie.ef.sak.behandling.domain.Behandling
+import no.nav.familie.ef.sak.behandling.dto.BehandlingDto
+import no.nav.familie.ef.sak.behandling.dto.tilDto
+import no.nav.familie.ef.sak.fagsak.FagsakService
+import no.nav.familie.ef.sak.fagsak.domain.Fagsak
+import no.nav.familie.ef.sak.infrastruktur.exception.Feil
+import no.nav.familie.ef.sak.infrastruktur.exception.feilHvis
+import no.nav.familie.ef.sak.infrastruktur.sikkerhet.SikkerhetContext
+import no.nav.familie.ef.sak.opplysninger.personopplysninger.GrunnlagsdataService
+import no.nav.familie.ef.sak.opplysninger.personopplysninger.pdl.gjeldende
+import no.nav.familie.ef.sak.vilkår.VilkårType
+import no.nav.familie.ef.sak.vilkår.Vilkårsvurdering
+import no.nav.familie.ef.sak.vilkår.VilkårsvurderingRepository
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import java.util.UUID
+
+@Service
+class GjenbrukVilkårService(
+    private val behandlingService: BehandlingService,
+    private val fagsakService: FagsakService,
+    private val vilkårsvurderingRepository: VilkårsvurderingRepository,
+    private val grunnlagsdataService: GrunnlagsdataService,
+    private val barnService: BarnService
+) {
+
+    private val secureLogger = LoggerFactory.getLogger("secureLogger")
+
+    fun finnBehandlingerForGjenbruk(behandlingId: UUID): List<BehandlingDto> {
+        val fagsak: Fagsak = fagsakService.hentFagsakForBehandling(behandlingId)
+        val behandlingerForGjenbruk: List<Behandling> =
+            behandlingService.hentBehandlingerForGjenbrukAvVilkår(fagsak.fagsakPersonId)
+        val fagsaker: Map<UUID, Fagsak> =
+            behandlingerForGjenbruk.map { it.fagsakId }.distinct().associateWith { fagsakService.hentFagsak(it) }
+        return behandlingerForGjenbruk.map { it.tilDto(fagsaker.getValue(it.fagsakId).stønadstype) }
+    }
+
+    @Transactional
+    fun gjenbrukInngangsvilkårVurderinger(
```

Comment:
Fint å flytte de til egen service, men hadde vært fint med tester for denne.

Eks tester som tar med vilkår med og uten barn

Reply:
fixd [49aa7d7](https://github.com/navikt/familie-ef-sak/pull/1749/commits/49aa7d7d505d391d7873cf9351372367dbf1165e)